packages/wallet/src/services/ActiveStakePublicKeysTracker.ts

```diff
@@ -0,0 +1,67 @@
+import { AccountKeyDerivationPath, AsyncKeyAgent, GroupedAddress } from '@cardano-sdk/key-management';
+import { Cardano } from '@cardano-sdk/core';
+import { Observable, OperatorFunction, distinctUntilChanged, from, map, mergeMap, switchMap } from 'rxjs';
+import { TrackerSubject } from '@cardano-sdk/util-rxjs';
+
+export interface ActivePubStakeKeysProps {
+  addresses$: Observable<GroupedAddress[]>;
+  rewardAccounts$: Observable<Cardano.RewardAccountInfo[]>;
+  keyAgent: AsyncKeyAgent;
+}
+
+const registeredRewardAccounts: OperatorFunction<Cardano.RewardAccountInfo[], Cardano.RewardAccount[]> = (
+  source$: Observable<Cardano.RewardAccountInfo[]>
+) =>
+  source$.pipe(
+    map((accts: Cardano.RewardAccountInfo[]) =>
+      accts
+        .filter(
+          (acct) =>
+            acct.keyStatus === Cardano.StakeKeyStatus.Registered ||
+            acct.keyStatus === Cardano.StakeKeyStatus.Registering
+        )
+        .map(({ address }) => address)
+    )
+  );
+
+const stakeKeyDerivationPaths =
+  (addresses$: Observable<GroupedAddress[]>) => (source$: Observable<Cardano.RewardAccount[]>) =>
+    source$.pipe(
+      switchMap((rewardAcctAddresses) =>
+        addresses$.pipe(
+          // Get grouped addresses of the active reward accounts
+          map((groupedAddresses) =>
+            groupedAddresses.filter((groupedAddr) => rewardAcctAddresses.includes(groupedAddr.rewardAccount))
+          ),
+          // Only interested in the stakeKeyDerivationPath
+          map((addresses) => addresses.map(({ stakeKeyDerivationPath }) => ({ stakeKeyDerivationPath }))),
+          // Trick to make Typescript understand the filter's result will be defined derivation paths
+          map((accountAndDerivationPaths) =>
+            accountAndDerivationPaths.filter(
```

Comment:
Won't we get duplicates?
```suggestion
            uniqBy(accountAndDerivationPaths.filter(
```

Reply:
Updated to use rewardAccounts as source, to avoid duplicates: [24705ec07](https://github.com/input-output-hk/cardano-js-sdk/commit/24705ec073ab39fbcc26a30d6332e03bc2d68c99)